File: 9. Palindrome Number/solution.cpp
/*
16 ms
5.9 MB

TC : O(N)
SC : O(1)
*/

/**
 * @author : SahilK-027
 * @brief : Using the logic of reverse an integer
*/

class Solution {
public:
    bool isPalindrome(int x) {
        long long reverse_num = 0;
        int temp=x;
        if(x<0)
           return false;
        while(temp!=0)
        {
            reverse_num = reverse_num*10+temp % 10;
            temp=temp/10;
        }
        return (reverse_num == x);
    }
};

File: 9. Palindrome Number/solution.js
/*
200 ms
50.4 MB

TC: O(n)
sc: O(1)

@author: SahilK-027
@brief: Iterative
*/


/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
    if(x < 0) return false;
    let reverse_x = 0;
    let num = x;
    while(num){
        let last_digit = num % 10;
        reverse_x = reverse_x * 10  + last_digit;
        num = parseInt(num / 10);
    }
    return (reverse_x === x);
};
